sympy/stats/drv_types.py

```diff
@@ -207,6 +207,8 @@ def check(p):
         _value_check((0 < p, p <= 1), "p must be between 0 and 1")
 
     def pdf(self, k):
+        if not k.is_integer or k < 1:
+            return S.Zero  # Return 0 for invalid values
         return (1 - self.p)**(k - 1) * self.p
```

Comment:
As I explained in the issue I don't think that the pdf method should do this. It should return a formula that is valid for valid values of `k`.

Testing `k < 1` will also fail for symbolic inputs so it would need to be a Piecewise but again I don't think it should be a Piecewise.